frontend/components/Page.js

```diff
@@ -0,0 +1,66 @@
+import React, { useContext, useEffect, useState } from 'react'
+import { string, object, bool } from 'prop-types'
+
+import Head from './Head'
+import NavBar from './NavBar'
+import { Container, Dimmer, Loader } from 'semantic-ui-react'
+import AuthProvider from '../contexts/AuthContext'
+import { useRouter } from 'next/router'
+
+const Content = ({ reqSignIn, reqSignOut, children }) => {
+  const { push } = useRouter()
+  const [loading, setLoading] = useState(true)
+
+  useEffect(() => {
+    const isAuthenticated = window.session?.user !== null
+
+    if (reqSignIn && !isAuthenticated) {
+      push('/login').then()
+    } else if (reqSignOut && isAuthenticated) {
+      push('/').then()
+    } else {
+      setLoading(false)
+    }
+  }, [loading])
+
+  if (loading) {
+    return (
+      <Dimmer active inverted>
+        <Loader>Loading...</Loader>
+      </Dimmer>
+    )
+  } else {
+    return <Container style={{ marginTop: 30 }}>{children}</Container>
+  }
+}
```

Comment:
Just found out this is the cause of the loading spinner. We need a better solution for this.

Reply:
See 5cbd357, 7d59eb1, cd36ff5, 50f395c. I solved this in `import-projects` as it was part of another problem.